### ISSUE TITLE
Meta: Resources

### ISSUE THREAD

**Issue by seanmonstar** [tokio-rs/console#39]
This is more of a meta issue about the concept of Resources. The Console will likely have two main concepts, Tasks and Resources (there could probably be more, smaller ones). Besides knowing about all the tasks that have been spawned in the runtime, it's also helpful to understand what _resources_ that task is interacting with, waiting on, moving around, and dropping. Resources could be sockets, files, timers, channels, mutexes, semaphores, and the like.

When inspect a task in more detail, we want to be able to view what resources the task has been touching, which it is waiting on, and if possible, the relationship with another task that could be unblocked by this one acting on the resource (sending a channel, unlocking a mutex, etc).

We also would like a resources view, similar to the tasks view, which can show us a list of the resources that currently are "alive", what their state is, and what tasks are waiting on them. And then just like tasks, a way to inspect an individual resource for more detail.

This will require a few steps (hence why this is a meta issue):

- [ ] Defining the names of the spans and/or structured fields to identify resources. (https://github.com/tokio-rs/tokio/issues/3954)
- [ ] Instrumenting Tokio's "resource" types to create a span when constructed, and to enter the span in their respective `poll_*` methods. (#40)
- [ ] Collecting this data in the Subscriber.
- [x] #63
